packages/components/image.tsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { connect, Head } from "frontity";
+import { Connect, Package } from "frontity/types";
+import useInView from "@frontity/hooks/use-in-view";
+
+// Hides any image rendered by this component that is not
+// inside a <noscript> when JS is disabled.
+const noJsStyles = `
+  :not(noscript) > .frontity-lazy-image {
+    display: none;
+  }
+`;
+
+// Finds all the images rendered by this component and maps
+// their `data-src` and `data-srcset` attributes to `src` and `srcset`
+// when the browser doesn't support Proxy or IntersectionObserver.
+const noProxyScript = `
+  if (typeof window !== "undefined" && (!("Proxy" in window) || !("IntersectionObserver" in window))) {
+    document.addEventListener("DOMContentLoaded", () => {
+      const images = document.querySelectorAll("img.frontity-lazy-image");
+      for (image of images) {
+        image.setAttribute("src", image.getAttribute("data-src"));
+        image.setAttribute("srcset", image.getAttribute("data-srcset"));
+        image.removeAttribute("data-src");
+        image.removeAttribute("data-srcset");
+        image.removeAttribute("style");
+      }
+    });
+  }
+`;
+
+interface Props {
+  src?: string;
+  srcSet?: string;
+  alt?: string;
+  className?: string;
+  loading?: "auto" | "lazy" | "eager";
+}
+
+type Image = React.FC<Connect<Package, Props>>;
```

Comment:
Why do you need `Connect` here. Is it because in the future the component will be connected?

Reply:
Yes, I forgot to remove it. Initially it was there because I was already implementing logic with the state and the media `id`, but then we decided to go without it for this first version.

Reply:
Ok. Let's leave it there then. Don't worry.